service/controller/v12/resource/instance/template/vmss.json

```diff
@@ -181,7 +181,10 @@
       "name":"[parameters('vmssName')]",
       "location":"[parameters('location')]",
       "tags":{
-        "provider":"[toUpper(parameters('GiantSwarmTags').provider)]"
+        "provider":"[toUpper(parameters('GiantSwarmTags').provider)]",
+        "k8s.io/cluster-autoscaler/enabled":"true",
+        "[concat('k8s.io/cluster-autoscaler/', parameters('vmssName'))]": "true"
```

Comment:
Did you test it? Never saw an expression used as a key in the template (only as value).
I think it should work though.

Reply:
I wasn't sure either. I tested it using this endpoint https://docs.microsoft.com/en-us/rest/api/resources/deployments/validate.

Reply:
What about testing it in the field?
We can do it together if you want.

Reply:
You can also check that the outcome is what you actually need it to be (i.e. that the tags are really there in the deployment)

Reply:
When deploying it for real, I got the following errors
```
The tag names 'k8s.io/cluster-autoscaler/enabled, k8s.io/cluster-autoscaler/i9ssh-master' have reserved characters '<,>,%,&,\\\\,?,/' or control characters. These characters are only allowed for tags that start with the prefix 'hidden, link'.
```

Which tags should we use? Since any tag can be used, we could just add a `name` tag with the current name, and another one to indicate that the cluster is being auto scaled. Something like
```name: i9ssh
cluster-autoscaler-enabled: true
```

WDYT?

Reply:
I'd use the `cluster-autoscaled-` prefix for the name as well in order to avoid future changes to give a different meaning to the  `name` tag given it's generic name and break autoscaling.

Reply:
done

Reply:
It works now

![image](https://user-images.githubusercontent.com/627038/69049590-8cfead80-0a00-11ea-8a2b-dcc4a7fad98e.png)